model-server/src/main/kotlin/org/modelix/model/server/handlers/IRepositoriesManager.kt

```diff
@@ -17,46 +17,77 @@
 package org.modelix.model.server.handlers
 
 import org.modelix.model.lazy.BranchReference
+import org.modelix.model.lazy.CLTree
 import org.modelix.model.lazy.CLVersion
 import org.modelix.model.lazy.RepositoryId
 
+/**
+ * Exposes high-level methods to interact with repositories, their branches and their versions.
+ *
+ * All methods are supposed to use non-blocking IO (or dispatch it appropriately).
+ */
 interface IRepositoriesManager {
+    suspend fun getRepositories(): Set<RepositoryId>
+
     /**
-     * Used to retrieve the server ID. If needed, the server ID is created and stored.
-     *
-     * If a server ID was not created yet, it is generated and saved in the database.
-     * It gets stored under the current and all legacy database keys.
-     *
-     * If the server ID was created previously but is only stored under a legacy database key,
-     * it also gets stored under the current and all legacy database keys.
+     * Creates a repository or throws an [RepositoryAlreadyExistsException].
```

Comment:
> This lacks some documentation now that we touch the code ;-)

> Why is the username needed here?
> Why can it be nullable?
> What is useRoleIds?

Reply:
Thanks, I think nowadays even the kotlin documentation has shifted towards using kdoc tags such as `@param` and `@throws` and I think we have mostly followed that approach in other places, too.

> [useRoleIds] is used when creating the [CLTree] for the initial version.

That still doesn't explain what that does actually.

Reply:
Done (using Block Tags)
Done (explaining useRoleIds)